### PR TITLE
Added active storage and validator

### PR DESCRIPTION
**Pull request by sergiomarrocoli**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,8 +34,10 @@ gem 'httparty'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
-# Use ActiveStorage variant
-# gem 'mini_magick', '~> 4.8'
+# ActiveStorage
+gem 'mini_magick', '~> 4.8' # Use ActiveStorage variant
+gem "aws-sdk-s3", require: false # Use S3 bucket for document storage
```
Comment (yndajas) [inline Gemfile:39]: Single quotes for consistency?

File: app/models/commitment.rb [hunk 2]
```diff
@@ -10,7 +10,13 @@ class Commitment < ApplicationRecord
   import_by objectives: :name
   has_and_belongs_to_many :governance_types
   import_by governance_types: :name
+  
+  has_many :progress_documents
 
+  has_one_attached :spatial_data
+  validates :spatial_data, 
+    content_type: %w(application/vnd.google-earth.kml+xml application/vnd.google-earth.kmz application/zip), 
+    size: { less_than: 25.megabytes }
```
Comment (yndajas) [inline app/models/commitment.rb:19]: No space between `has_one_attached` and the related `validates` here, in contrast with the spacing in `progress_document`

File: app/models/progress_document.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class ProgressDocument < ApplicationRecord
+  belongs_to :commitment
+
+  has_one_attached :document
+
+  validates :document, attached: true, content_type: %w(application/msword application/pdf), size: { less_than: 25.megabytes }
```
Comment (yndajas) [inline app/models/progress_document.rb:6]: Different spacing to `commitment`, as noted above

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+aKrv1m3B8xtqHq1Rg3f3ffwnLU3u7XLTW7fkmmq3W7/jGacYhEhnTXrp209p5u2IGlrhbxtpYi8vqXbA4kpqJSyCSaBrZuWR+2+r0xpKNhO7A00mC56mKcivNIIlOvably1zgyJqj3PNz/s+3CDz8YC/YzXOvMioFH1nafh3woj1aQ9jI3rYXGO5TU+23ze9ylhbLttXaSiIHxRLdA06kmAL271NbyGtiXFbexwbNCxP9gDxYawliWwA4riUeZGwVqNtfxBUoJ/cekNbnp1sq6RkebNAiUmn0wCbhn2TFvjk1Ym6tXDe6x7wKV3iMThX0WFe2j24g6RkltH060uFxDbZxjec+qPGcVjeyAKzL6DCTO0ak+HqpKe1hh3sxhg4GYAKAOJgusjooS8u6g15RWNcWFmNVD+sOt4/q0kPepEQdPnY1N1O9VrwPHhL018JVlAXfZRzTwrfmlAgbN0kqII9DP6W9dYqFpAopUUjre1/fMS28XKmHZszY+PC2PFajMDMzIkBd92FB6X/ZPqZZAJD+R+ZmTzmiDQBrm6kvcVQu0R1i0Sbd0Sp5WuHWleyOnhV7EQR9fuZ7b1/gXUhWR+kBakNVSMdYJsYxYdIvpD19MqTKc9qQK685iUncE5I+sc5Bn0SyYRHzUedpiSGgWkzCUNV0+aCaTswVcQ6WjsBJWcsrEOK8UrMs8+gILOjquOGwg9h5vNzRagCI4msU/dxf2QVMfk9djFXegj4rTTGZ3/3AF1bTi0yXan8EZ+ydy79mDm1/RHYJnRn0miopAqO51zKPj+IuoN3qXI230ccz6wyA+Iksf1DEYYLugQLhlwLJl29Put/vlkLSH/dqO10xst6MxEeanM2TMBQ9l4tokHtOlbxYiZpxFP/WaMkAwEs1jJfc4xobm4TdyDS1YhTs3WDlRQcnp8L/ngzU14dbIZpQTHBlvEq2xXGh4jKDxvePjB/KUagw0k4AKtacmwzcaerA7TQaZFwBh0TdFtygmUBLNImQ1X8tQ3w2r+BYXkGVKil1UUqTmgrWhKlvRocEH3gB+xfNH0tFa2LSczZE08Zgy4cDIh7qEf9Sfhdk8kt7n7uPcFvDShU06xxnq+hdjon5hc=--EhOO/cIuwO3gyLCe--E9//qEVdA8omSAl/Im5+pA==
```
Comment (yndajas) [inline config/credentials.yml.enc:1]: Does this file need to be added to .gitignore and moved to LastPass, or is it safe to keep public?

File: config/locales/en.yml [hunk 5]
```diff
@@ -34,6 +34,9 @@ en:
 
   errors:
     format: "%{message}"
+    messages:
+      content_type_invalid: "has an invalid content type: %{content_type}"
+      file_size_out_of_range: "size %{file_size} is not between required range"
```
Comment (yndajas) [inline config/locales/en.yml:39]: Should this be simplified to "file size is too large", since we aren't specifying a minimum?

**Review by yndajas — CHANGES_REQUESTED** [review]
Just a few formatting things, a query on the credentials and another on error message text

Guessing there's not much we can test yet, but `rails s` still serves the site fine after running the migrations

File: config/credentials.yml.enc (hunk 4 above)
Comment (sergiomarrocoli) [reply config/credentials.yml.enc:1]: credentials are kept in the repo. they're decrypted with the config/master.key, which is in the gitignore, so they are safe. we keep the master.key in LastPass so we all use the same one. i haven't done that though :-) will do it now

File: config/locales/en.yml (hunk 5 above)
Comment (sergiomarrocoli) [reply config/locales/en.yml:39]: chnaged to "file size %{file_size} is too large".

File: app/models/progress_document.rb (hunk 3 above)
Comment (sergiomarrocoli) [reply app/models/progress_document.rb:5]: moved. not sure if this is right? happy to change it again if not

**Review by yndajas — APPROVED** [review]

**Pull request merged**